src/map.rs

```diff
@@ -1253,6 +1253,13 @@ where
         self.into_iter()
     }
 
+    /// Reverses the order of the map’s key-value pairs in place.
+    ///
+    /// Computes in **O(n)** time and **O(1)** space.
```

Comment:
O(1) space isn't true when we're calling `save_hash_index` because that's O(n) -- but the optimization we discussed will avoid the need for that.

The idea is to mimic `restore_hash_map`'s `apply_new_index`, but changing this:

```rust
    pos.set_pos::<Sz>(new_index[i]);
```

We know what the new reversed index will be without a saved table, just:

```rust
    pos.set_pos::<Sz>(len - i);
```

Reply:
Pardon me -- that should be `len - i - 1`!

Reply:
I think I'm doing something wrong: using both `self.entries.len()` and `self.indices.len()` as `len` gives me out of bounds errors.

Reply:
Aha, that was the cause, curse you off-by-one errors! Using `self.entries.len() - i - 1` works wonderfully.

Reply:
Yeah, `self.entries.len()` is correct, but I was off by one for the new index.